sound/soc/intel/boards/sof_sdw_common.h

```diff
@@ -56,15 +56,18 @@ enum {
 #define SOF_SDW_CODEC_TYPE_AMP		1
 #define SOF_SDW_CODEC_TYPE_MIC		2
 
+#define SOF_SDW_CODEC_MAX_DAI_NUM	3
+
 struct sof_sdw_codec_info {
 	const int part_id;
 	const int version_id;
 	const int codec_type;
 	int amp_num;
 	const u8 acpi_id[ACPI_ID_LEN];
-	const bool direction[2]; // playback & capture support
+	const bool direction[SOF_SDW_CODEC_MAX_DAI_NUM][2]; /* playback & capture support for each dai */
```

Comment:
this is a bit misleading, SoundWire DAIs are not bi-directional.

Reply:
> this is a bit misleading, SoundWire DAIs are not bi-directional.

We have the same thought @plbossart My first version is to change it to `const int direction[SOF_SDW_CODEC_MAX_DAI_NUM]` to indicate the direction for each dai. But `direction` is also used by SSP DAIs. See
[playback = info->direction[SNDRV_PCM_STREAM_PLAYBACK];
capture = info->direction[SNDRV_PCM_STREAM_CAPTURE];](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw.c#L1379)